### PR TITLE
test: try annotated tags with https://github.com/jenkins-infra/jenkins-maven-cd-action/pull/26

### PR DESCRIPTION
**Pull request by dduportal**
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: .github/workflows/maven-cd.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+name: maven-cd
+on:
+  workflow_call:
+    secrets:
+      MAVEN_USERNAME:
+        required: true
+        description: Maven username used for deploying the plugin jar to Jenkins Artifactory Repository
+      MAVEN_TOKEN:
+        required: true
+        description: Maven token used for deploying the plugin jar to Jenkins Artifactory Repository
```
[the remark this answers is not in the document]
Comment (dduportal) [reply .github/workflows/maven-cd.yml:12]: 🤦

[the remark this answers is not in the document]
Comment (dduportal) [reply .github/workflows/maven-cd.yml:12]: Mmmh 🤔 https://github.com/jenkinsci/jenkins-infra-test-plugin/blob/feat/annotated-tags/.github/workflows/cd.yml#L13 the current `cd` is calling it.

It's only a "copy" of the workflow from jenkins-infra/github-reusable-workflows, except it's in the same repository + uses my fork of the maven-cd for the step release.

[the remark this answers is not in the document]
Comment (timja) [reply .github/workflows/maven-cd.yml:12]: try it out no harm anyway, just merge here and iterate how you need

File: .github/workflows/maven-cd.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        fetch-depth: 0
+
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 11
+
+    - name: Release
+      uses: dduportal/jenkins-maven-cd-action@v1
```
Comment (jglick) [inline .github/workflows/maven-cd.yml:63]: I think you want to use a branch name rather than a tag to test?

Comment (dduportal) [reply .github/workflows/maven-cd.yml:63]: Oh I did not know branch could be used for the actions specifications

**Pull request merged**